src/components/carousel/daikin-carousel.ts

```diff
@@ -0,0 +1,334 @@
+  currentIndex = 0;
+
+  /**
+   * Specify the variant of the control button.
+   */
+  @property({
+    type: String,
+    reflect: true,
+    attribute: "control-button-variant",
+  })
+  controlButtonVariant: "ghost" | "fill" = "ghost";
+
+  /**
+   * Whether or not to allow content to be moved by swiping.
+   */
+  @property({ type: Boolean, reflect: true, attribute: "allow-swipe" })
+  allowSwipe: boolean = false;
+
+  @queryAssignedElements({ selector: "daikin-carousel-item" })
+  private readonly _items!: readonly DaikinCarouselItem[];
+
+  @state()
+  private _itemCount: number = 0;
+
+  @state()
+  private _isSwipe = false;
+
+  @state()
+  private _swipeX = 0;
+
+  private _swipeStartX = 0;
+  private _swipeEndX = 0;
+
+  private _prevButton = createRef<HTMLButtonElement>();
+  private _nextButton = createRef<HTMLButtonElement>();
+  private _indicatorButtonsContainer = createRef<HTMLElement>();
+
+  private _emitSelect(
+    operation: "prev" | "next" | "indicator",
+    beforeCurrentIndex: number
+  ) {
+    this.dispatchEvent(
+      new CustomEvent("select", {
+        detail: { operation, beforeCurrentIndex },
+      })
+    );
+  }
+
+  private _handleClickIndicator(index: number) {
+    const beforeCurrentIndex = this.currentIndex;
+
+    this.currentIndex = index;
+    this._emitSelect("indicator", beforeCurrentIndex);
+  }
+
+  private _handleKeydownIndicator(event: KeyboardEvent): void {
+    const moveOffset = (
+      {
+        ArrowRight: 1,
+        ArrowLeft: -1,
+      } as const
+    )[event.key];
+
+    if (!moveOffset) {
+      return;
+    }
+
+    const buttons = [
+      ...(this._indicatorButtonsContainer.value?.querySelectorAll("button") ??
+        []),
+    ];
+
+    this._moveBy(moveOffset);
+    buttons[this.currentIndex].focus();
+  }
+
+  private _handleSlotchange() {
+    this._updateCounter();
+    this._updateItemLabel();
+    this._updateItemActive();
+  }
+
+  private _handleTouchstart(event: TouchEvent) {
+    this._swipeStartX = event.touches[0].pageX;
+    this._isSwipe = true;
+  }
+
+  private _handleTouchmove(event: TouchEvent) {
+    this._swipeEndX = event.changedTouches[0].pageX;
+    this._swipeX = this._swipeStartX - this._swipeEndX;
+  }
+
+  private _handleTouchend() {
+    this._isSwipe = false;
+
+    // If the interval between touch operations is extremely short,
+    // it is determined to be an erroneous operation and the process is terminated.
+    if (Math.abs(this._swipeX) < 10) {
+      return;
+    }
+
+    this._moveBy(Math.sign(this._swipeX) as 1 | -1);
+    this._swipeX = 0;
+  }
+
+  private _moveBy(moveOffset: 1 | -1) {
+    if (
+      (moveOffset === -1 && this.currentIndex <= 0) ||
+      (moveOffset === 1 && this.currentIndex >= this._itemCount - 1)
+    ) {
+      return;
+    }
+
+    const beforeCurrentIndex = this.currentIndex;
+
+    this.currentIndex = this.currentIndex + moveOffset;
+    this._emitSelect(moveOffset === 1 ? "next" : "prev", beforeCurrentIndex);
```

Comment:
nit. When performing this kind of validation, it is safer to calculate the values that are scheduled to be changed in advance and determine whether they are appropriate or not.

```suggestion
    const newIndex = this.currentIndex + moveOffset;
    if (newIndex < 0 || newIndex >= this._itemCount) {
      return;
    }

    const oldIndex = this.currentIndex;
    this.currentIndex = newIndex;
    this._emitSelect(moveOffset === 1 ? "next" : "prev", oldIndex);
```